New module - semodule

bandit145 opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
Adds an semodule Ansible module for compiling and applying .te (Type enforcement files) as discussed in https://github.com/ansible/ansible/issues/27349

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- New Module Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
semodule

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
Most of the logic for controlling this module is written in the action, this seemed like a good way to cut down on needing to copy the .te file to the target host if not needed.
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below
.te file:
module ansible-semodule-test 1.0;

require {
	type proc_net_t;
	type sshd_t;
	class file { open read };
}

allow sshd_t proc_net_t:file { open read};

- name: install test module
  semodule:
    src: test.te

TASK [semodule : install test module] *********************************
changed: [testhost] => {"changed": true, "name": "ansible-semodule-test", "version": "1.0"}

TASK [semodule : debug] ********************************************************
ok: [testhost] => {
    "test1_mod_install": {
        "changed": true, 
        "failed": false, 
        "name": "ansible-semodule-test", 
        "version": "1.0"
    }
}
```

bandit145 committed:
writing integration tests

bandit145 committed:
basic working smeodule action+module

bandit145 committed:
addded checksumming and module utils

bandit145 committed:
added check mode to action and integration tests finished

bandit145 committed:
unit tests

bandit145 committed:
added documentation removed unneeded function

bandit145 committed:
fixed import and formatting errors

bandit145 committed:
fixed metadata spelling error

bandit145 committed:
fixed test_semodule pep8 issues and made metadata_version a string

bandit145 committed:
added alias for integration test

bandit145 commented:
Due to investigation prompted by failing rhel8 and fedora tests it seems newer versions of semodule do not display module versions any-more, as such I will  drop "latest" support and operate only off module name info.

Relevant bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1690779

bandit145 committed:
removed semodule version checking

bandit145 committed:
removed uneeded test step

bandit145 committed:
removed latest from choices

bandit145 committed:
added parameter checks in action and fixed AnsibleErrors. supports opensuse integration test

bandit145 committed:
deleted broken test

bandit145 committed:
skip ubuntu and skip opensuse15py2 (selinux-policy requires python3 deps)

bandit145 committed:
added granular skips for ubuntu and opensuse15py2

bandit145 committed:
updated for new sanity checks

goneri commented on lib/ansible/modules/system/semodule.py:
```diff
@@ -0,0 +1,132 @@
+short_description: Manage selinux policy modules
+description:
+  - Use this module to manage selinux policy modules from Type Enforcement files
+version_added: "2.9"
```
Hi @bandit145, You need to bump this to 2.10 now.

bandit145 replied:
Will do.

bandit145 committed:
updated version added

bandit145 commented:
@goneri Can you review this again? Thanks.

abadger committed:
Add a sanity test for yaml to make sure we can parse all yaml

Check that all yaml we ship is parsable by the pyyaml c backend.  Since
Ansible uses Pyyaml for docs and playbooks, if the yaml files aren't
parsable, they will error out if they were used.

Warn and skip yamllint if libyaml backend is not present

Ignore new errors in examples until someone can fix them

abadger committed:
Change the check_pyyaml() function to return the features of pyyaml that we check

mariolenz committed:
VMware: Add advanced settings to vmware_cluster_ha

goneri committed:
prepare_vmware_tests: add scenario with one ESXi

Add the ability to test with just one vcenter instance and a single
ESXi. This allow us to cover 90 modules.

goneri committed:
vmware/test: init esxi_user with esxi1_username

By default, `esxi_user` is initialize as `root`. We now use `esxi1_username`
instead.

felixfontein committed:
Add notes on docker_network connectivity for containers and services. (#65840)

billdodd committed:
Fix version param in deprecate() call and update porting guide (#65894)

* fix version= param in deprecate() call and update porting guide

* add changelog fragment

haiyuazhang committed:
add image reference in vmss instance info (#65872)

haiyuazhang committed:
azure_rm_virtualmachine: fixing issue #63844 (#65601)

mnecas committed:
ovirt_snapshot: select disks (#65729)

mnecas committed:
ovirt_network: correct external provider (#65701)

* ovirt_network: update external provider

* unnecessary remove blank line

* remove update check for external_provider

* first try to import then create

mnecas committed:
ovirt_affinity_group: add host rule option disabled (#65704)

Akasurde committed:
vars: Fail with warning for extra_vars filename without @ sign (#59915)

Rather than silently processing extra_vars filename without @ sign,
CLI now fails with appropriate warning about requirement.

Fixes: #51857

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

goneri committed:
vmware/test: more modular teardown.yml

Some teardown tasks are now enabled only if we have ESXi in the configuration
file. On my system and in a vcenter only scenario, `vcenter_folder` goes from
42.7 to 31.2s.

ivog74 committed:
Add ansible cli options --ask-vault-password and --vault-pass-file (#63782)

* Move new Ansible cli options '--ask-vault-password' and '--vault-pass-file' to the existing calls to add_argument
* Add changelog fragement
* Change order of ansible cli arguments to use --ask-vault-password and --vault-password-file by default
* Update runme.sh in vault integration tests to test new options --ask-vault-password and --vault-pass-file

internetionals committed:
Fix 'Assert CNAME failure' assertion (#65875)

waffie1 committed:
Cloudformation info checkmode (#65958)

* set supports_check_mode=True

I don't see any reason why this module cannot run in check mode.  The following API calls are made, none of which modify configurations

All using boto3.client('cloudformation')
describe_change_set
describe_stacks
describe_stack_events
get_paginator
get_stack_policy
get_template
list_change_sets
list_stack_resources

* duplicate cloudformation_info tasks with checkmode

Duplicated all existing cloudformation_info tests and added check_mode: yes to them

* delete duplicate empty line

SpamapS committed:
Add support for ECR Lifecycle Policies to ecs_ecr (#48997)

* Fix copy/pasta for ecs_ecr test names

* Add support for lifecycle policies to ecs_ecr

New feature for ecs_ecr to support [ECR Lifecycle Policies][].

Fixes #32003

 [ECR Lifecycle Policies]: https://docs.aws.amazon.com/AmazonECR/latest/userguide/LifecyclePolicies.html

* Improve error message for ecs_ecr parsing errors

Replaces the exception and stack trace with a description of what's
actually going wrong from a user perspective.

* Rename delete policy to purge policy

Marks the `delete_policy` parameter as deprecated, to be removed in
Ansible 2.6.

* Add version_added to purge_policy

* Remove changing results based on verbosity

What I really want is --diff support, and changing results based on
verbosity is abnormal.

* Ensure repository name is lowercase

* Fix deprecation cycle to 4 releases

* Use a YAML anchor for credentials

* Remove filters from assertions

* Add minimal permissions needed

* Updating version_added and deprecation cycle

The original PR sat while a few releases happened.

* Bumping version added and deprecation version

We missed the 2.8 release.

* Removing bare except:

This is not allowed and is generally bad practice.

* Fix lint errors

* update ansible release metadata

* Use the new alias deprecation scheme

This was added in the time the PR has been in development, so rework
things to use it.

* Add test coverage

This makes sure that lifecycle_policy is produced when passed in.

*Also a minor suggestion for simplification from PR.

* Restore changes from 62871 lost in rebase

* Add changelog

* Remove version_added for new purge_policy option

Per sanity test fail.

s-hertel committed:
free strategy - include failed hosts that were notified (#65576)

* free strategy - include failed hosts that were notified so --force-handlers is used

* trim line length a bit

* Loop over the force handler tests with the strategies linear and free

* rename changelog

* Use the play iterator instead of TQM for accurate failure representation in blocks

* Remove hack in a backwards compatible way for 3rd party plugins

bmillemathias committed:
Set a deprecation warning and version for option sha256sum (#65637)

The option sha256sum was declared deprecated more than 2 years ago without
setting a removal version. Let's do that now.

fixes #65636

bmillemathias committed:
[get_url] Return status_code on HTTP 304 (#65739)

Return field status_code on HTTP status 304 (implemented by lkthomas)
Add an integration test for this case.
Fixes #65263

mulatinho committed:
elb_{network_lb,target_group}: allow UDP/TCP_UDP (#65828)

elb_network_lb.py: allow UDP and TCP_UDP protocols

- Fixing documentation
- Add support to UDP and TCP_UDP as described on AWS SDK

elb_target_group.py: allow UDP, TLS, TCP_UDP proto

- Fixing documentation
- Making health checks with response codes and paths only in HTTP/HTTPS
- Allow UDP, TLS, TCP_UDP protocols as described on AWS SDK.

others:
- Added changelog fragments
- Integration test

Fixes: #65265
Signed-off-by: Alexandre Mulatinho <alex@mulatinho.net>

tremble committed:
fix ec2_vpc_net test instability (#65184)

* Move ec2_vpc_net tests to group1, group2 is running much longer

* Allow for VPC CIDRs to be "associat*ing*", things in AWS are "eventually consistent" and occasionally take longer than we would like

ngp-star committed:
VMware: filter VMs of Same name on the basis of folder (#65922)

vmware_guest now filters VMs of same name on the basis of additional information of folder.

sabre1041 committed:
VMware: Corrected documentation on vmware_guest module (#65733)

Update link for vapp_properties as per VMware docs in vmware_guest module

ganeshrn committed:
Fix netconf plugin related to collections (#65718)

* Fix netconf plugin related to collections

Fixes #65655 (partly)

*  Make netconf plugins configurable so that the
   information of ncclient device handler
   for give platform resides in the platform
   specific netconf plugin.
*  If the device handler value in ncclient is
   different from the ansible_network_os value
   the right value of `ncclient_device_handler`
   should be set in the plugin documentation.

* Fix review comments

* Fix CI issue

* Fix review comment

s-hertel committed:
Remove illusion of caching in the inventory script plugin (#64151)

* Load the cache plugin for the inventory script plugin and fix update cache logic

* Remove the illusion and let individual scripts deal with it

* reword that

* Deprecate instead of a hard failure

* changelog

mattclay committed:
Add ansible-test constraint to avoid coverage 5.0+

samdoran committed:
user - honor update_password parameter on BusyBox hosts (#65977)

The check for this parameter was missing from BusyBox.modify_user(), resulting in unexpected password changes to existing user accounts.

mattclay committed:
Disable failing azure_rm_storageaccount test.

pilou- committed:
clc_aa_policy: deprecate unused 'wait' parameter (#64367)

torvitas committed:
nmcli: Add compatibility for new networkmanager library (#65726)

Fixes: #48055
* add compatibility for new networkmanager library
* update debian/ubuntu docs
* move import gi into try

dano0b committed:
#58578 support absolute paths additionally (#58812)

* support absolute paths additionally

* add changelog

* rename changelog

* fix changelog to bugfix

dr4Ke committed:
terraform workspace list failure warns instead of fails (#65044)

* terraform workspace list failure warns instead of fails

Fixes #64788

* add a changelog fragment

atombrella committed:
Use context manager for file handling in misc files. (#65372)

The file handles are left open in those files.

stefanhorning committed:
Fix to return data when using lambda_info module (#64548)

* Fixed issue 64479 with lambda_info module

* Added integration tests for lambda_info module

* Moved lambda_info tests into already existing aws_lamda testsuite for easier test setup.

Co-authored-by: Jill R <4121322+jillr@users.noreply.github.com>

D3DeFi committed:
zabbix_hostmacro support for zabbix native macro names and macro_value not required when state=absent (#65884)

milouse committed:
Don't rely anymore on locale-dependent wording in Pacman output (#65878)

Use line position instead, as specified in upstream pacman source code.

See: https://git.archlinux.org/pacman.git/tree/src/pacman/package.c#n260
Fixes: https://github.com/ansible/ansible/issues/64682
Fixes: https://github.com/ansible/ansible/issues/65237

Akasurde committed:
Revert "Don't rely anymore on locale-dependent wording in Pacman output (#65878)" (#66029)

This reverts commit cd0e038b746f426003ba7177f5ff4882db9dfe9b.

sky-joker committed:
Include groups in name-based search results (#66026)

add changelogs

L3n41c committed:
Fix pacman stdout parsing in the Ansible module (#65238)

pacman output is localized and the Ansible module is parsing its output.
So, we need to force the locale.
Add changelog fragment

Fixes #65237

evgeni committed:
don't use `with_file` in exclusive=True examples (#65967)

geerlingguy committed:
Fixes #65952: Remove ambiguous example from k8s docs to clarify file usage. (#65953)

Commit:
fix AWS cloudfront purge_cache_behaviors misspelling (#65927)

tpraxl committed:
Be more specific about the requirements per distro (#65810)

D3DeFi committed:
improved zabbix action conditions documentation in zabbix_action module (#65808)

choeffer committed:
Update os_group_info.py (#65115)

abadger committed:
Add a banner which will display on the testing site. (#64763)

* Add a banner which will display on the testing site.

Thanks to shane for the javascript code

* change to more noticable color

Co-authored-by: Sandra McCann <samccann@redhat.com>

Akasurde committed:
VMware: Handle NoneType in vmware_vmkernel_info (#65834)

Check for virtualNicManager in Esxi host system before accessing properties in vmware_vmkernel_info.

Fixes: #62772

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
Check if task file is specified for import_tasks (#57572)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

trishnaguha committed:
fix nxos_hsrp keyerror (#65796)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

haiyuazhang committed:
fixing #64404 (#66032)

NilashishC committed:
Fix nxos_l3_interfaces rendering of dhcp (#66049)

Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

Akasurde committed:
VMware: Add example for creating VM without disks (#66057)

Fixes: #39743

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

abhaykadam committed:
Fix env lookup plugin error on utf8 values (#65541)

This commit fixes issue #65297.

The env lookup plugin used to fail when environment variable value
contained any UTF-8 characters (e.g., δ, ζ).

D3DeFi committed:
zabbix_host idempotency fix for situation where ZabbixAPI handles host groups case insensitively (#63860)

Akasurde committed:
homebrew_cask: Update documentation and examples (#66073)

* properly document ``upgrade`` option as an alias to ``upgrade_all``, and not a separate option
* Update documentation.
* Update examples.

Fixes: #34696

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

pypypyo14 committed:
Homebrew: some functions ignored check_mode option (#65387)

update_homebrew and upgrade_all now support check_mode

Fixes: #65352

Commit:
os_volume: optional bootable parameter for volume (#43945)

rohitthakur2590 committed:
allowing banner without stripping spaces (#62573)

Signed-off-by: rohitthakur2590 <rohitthakur2590@outlook.com>

mikewiebe committed:
Fix nxos_user roles bug (#65962)

andreaso committed:
nsupdate: Use provided TSIG key for all queries (#63174)

In addition to signing update queries also use the TSIG key to sign
lookup queries. By doing that we allow a hidden master to not only to
be looked down network wise, but also TSIG wise.

A bonus benefit of threating update queries and lookup queries more
the same is that will allow for all queries to be refactored into a
shared helper method. Currently we have a bit too much duplicated code
within the module.

justjais committed:
To fix the bug where IOS vlans module throws traceback (#64633)

* fix 64515

* add fix

agaffney committed:
Track source of vars

Brajen259 committed:
update aws_eks_cluster.py (#66105)

palvarez89 committed:
docker_swarm_service: Fix description of limits.memory (#66078)

This option corresponds to the '--limit-memory' option. Also
fix other mentions memory reservation instead of limit.

felixfontein committed:
docker_network: fix multiple subnet (of same IP version) idempotence (#65839)

* Fix multiple subnet (of same IP version) idempotence for docker_network.

* Add changelog.

* Unit tests no longer make sense, since the part of the code they test has been removed.

* Re-add CIDR validation. Move it to better position (module setup instead of idempotence check).

* Update changelog.

* Only run new tests on VM test images.

* Actually do what is documented. Especially since an empty object is a valid value for aux_addresses.

felixfontein committed:
docker_container: wait for removal if removal is in process (#65854)

* Allow to inspect containers directly.

* Wait for containers to be removed before recreating them.

* Also wait for containers to be removed before creating them.

* Add changelog.

Zhiwei-Dai committed:
Update OS_FAMILY_MAP for EulerOS (#66104)

Add support for EulerOS

Signed-off-by: Dai zhiwei daizhiwei3@huawei.com

Andersson007 committed:
mysql_db: fix doc formatting (#66087)

NilashishC committed:
Make autocost bandwidth calculation a floor division for Py3 compatibility (#66095)

Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

tremble committed:
AWS Guidelines: Make 'security_token' optional so that they can be run outside of CI (#66107)

xenlo committed:
Correct spelling mistake in proxmox_kvm (#66133)

* Correct a little spelling mistake in proxmox_kvm - changed 'update' to 'updated'

Andersson007 committed:
postgresql_query: refactoring (#66084)

hannseman committed:
docker_swarm_service: Fix parsing of Healthcheck.StartPeriod (#66151)

* Parse Healthcheck.StartPeriod properly

* Add changelog fragment

* Use proper markup in changelog

Co-Authored-By: Felix Fontein <felix@fontein.de>

Co-authored-by: Felix Fontein <felix@fontein.de>

GurTelem committed:
Fixed typo in vultur module (#66159)

Changed `exmaple` to `example`

felixfontein committed:
Don't wait for removal during check mode. (#66145)

qoreQyaS committed:
wording in examples changed ios to eos (#66131)

wrong module was used in examples: should be eos_vlans not ios_vlans

crossan007 committed:
Update Documentation for win_scheduled_task.py (#66079)

Passing an empty string through this parameter is not gracefully handled by the module implementation, so adjust the documentation to only include the `PT0S` method of specifying unbounded `execution_time_limit`. The win_scheduled_task execution_time_limit must be set explicitly to an empty string to permit unbounded execution time.

A newly created task without a value for `TaskSettings.ExecutionTimeLimit` will default to 72 hours (https://docs.microsoft.com/en-us/windows/win32/taskschd/tasksettings-executiontimelimit) even if you omit the Ansible task parameter `execution_time_limit`.

GurTelem committed:
Fix typo (#66158)

matt-telstra committed:
make concurrent aws_acm integration tests not clash (#64660)

* add uuid for aws_acm test resources, allow concurrent tests

* remove debug task in aws_acm int test

rohitthakur2590 committed:
vyos_config bug fix #66110 (#66122)

Signed-off-by: rohitthakur2590 <rohitthakur2590@outlook.com>

kbreit committed:
Correct fetch_url error code documentation (#66061)

HttpError is passed for 400 and above, not only greater than 400. Change docs to `>=` instead of `>`.

Akasurde committed:
lambda_info: Remove unused variable and import (#66124)

* Clean up residual code from #64548

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

nqb committed:
add example for 'search_paths' option (#65731)

credit to GoneWacko, see #51190 for reference

orescek committed:
vrong usage of name tag in example (#66129)

Docs: the capitalization of the tag should match in the description and the example. Change the example to "Name" (not "name") to match the description above in the efs_info module documentation.

modular-magician committed:
Bug fixes for GCP modules (#65792)

modular-magician committed:
Bug fixes for GCP modules (#65681)

modular-magician committed:
Bug fixes for GCP modules (#65597)

modular-magician committed:
Bug fixes for GCP modules (#65445)

modular-magician committed:
Bug fixes for GCP modules (#65309)

Andersson007 committed:
postgresql modules: use query parameters with cursor objects (#65862)

* postgresql modules: use query parameters with cursor objects

* add changelog fragment

* fix typo in changelog fragment

Andersson007 committed:
mysql_db: add master_data parameter (#66048)

* mysql_db: add master_data parameter

* add changelog

* fix CI

dukex committed:
PostgreSQL : Create unique index (#66157)

* Postgresql : Create unique index

TODO:

* fix syntax error

* add version added

* adding specs and improve doc

* add changelog

* Update lib/ansible/modules/database/postgresql/postgresql_idx.py

* Update lib/ansible/modules/database/postgresql/postgresql_idx.py

* remove check about concurrent

* Apply suggestions from code review

Akasurde committed:
Misc typo fixes (#66120)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

felixfontein committed:
docker_container: configure removal wait timeout (#66144)

* Add support for timeout while waiting for state.

* Allow to limit removal wait time.

* Add changelog.

* Forgot version_added.

* Add some check mode tests.

* Use removal_wait_timeout in tests.

markuman committed:
Assign AWS Cloudwatch Metric Filter via Ansible (#59925)

* #42292 first draft.

* fix some documentation

* use snake dict keynames

* return metric filter variable

* rework documentation

* try to fix documentation. fix condition when defaultValue is 0. update version string

* fix intenion, syntax and undefined

* try solve state documentation

* define type for state in doc

* rework integration test and fix minor typos

* add defaults for integration test

* update documentation, integration tests, and return value

* fix code formatting and documentation

* irgnore errors in ignore block

* Update lib/ansible/modules/cloud/amazon/cloudwatchlogs_log_group_metric_filter.py

Co-Authored-By: Jill R <4121322+jillr@users.noreply.github.com>

* add options to metric_transformation

* fix pep-8

Co-authored-by: Jill R <4121322+jillr@users.noreply.github.com>

thucke committed:
Add support for OSCM-distribution in hostname module (#66190)

Fixes: #66189

Added subclass to detect the new distribution OSMC

Ikuyadeu committed:
Fixed misc typo in netcup_dns module (#66201)

Changed 'authentification' to 'authentication'

rohitthakur2590 committed:
pre cleanup services (#66188)

Signed-off-by: rohitthakur2590 <rohitthakur2590@outlook.com>

Ikuyadeu committed:
name2Name for myTestNameTags (#66203)

heyleke committed:
Helm: add 'values' examples (#66197)

* DOC: add 'values' examples to helm module docs, see https://github.com/ansible/ansible/issues/31117

Co-authored-by: Abhijeet Kasurde <akasurde@redhat.com>

larsks committed:
Correct rst syntax in playbooks_filters.rst (#66198)

The `regex_replace` documentation included an example that was meant
to be preformatted text but due to an indentation error was rendering
incorrectly.

Andersson007 committed:
ali_instance: fix doc formatting (#66086)

samccann committed:
fixed some broken links (#66182)

rrey committed:
Fix callback set_options issue with collections (#66128)

Fixes: #66127

Toniob committed:
add a path_join filter which joins path components (#62713)

atombrella committed:
Use a context handler for file handling in paramiko_ssh.Connection. (#65624)

winem committed:
docker_container - #65993 - update restart policy (restart policy & restart retries) wit… (#66192)

* #65993 - update restart policy (restart policy & restart retries) without restarting the container

* - proper indentation on the continuation-line
- set restart_policy to the correct value independent from the api version

* - move restart_policy definitions into the if block
- add a new variable for the restart_policy configuration value

* add changelog fragment

* typo; minus -> underscore

* rename changelog fragment to contain the correct module name

* rename restart_policy_config_value to just restart_policy and refer to the correct dict values

samdoran committed:
Remove unstable from yum tests (#66218)

They seem to be stable again on Fedora 31.

njferrant committed:
vmware_guest: exclude dvswitch_name from guest os customization (#65997)

- exclude dvswitch_name from guest os customization
- add changelog fragment for ansible PR#65997

NilashishC committed:
Fix nxos_snmp_community command ordering (#66094)

Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

NilashishC committed:
Fix rendering duplicate peer-ip commands (#66088)

Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

NilashishC committed:
Temporarily disable vpc peer-link tests (#66121)

Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

NilashishC committed:
Use dict_diff instead of sets to take diff lag members (#66126)

Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

LKaemmerling committed:
Fix hcloud_server and hcloud_server_info module when the image is None (#66234)

oceyral committed:
pagerduty_alert: fix IndexError exception (#46770)

* pagerduty_alert: fix IndexError exception

* clean a bit

sivel committed:
Ensure that data within a tuple is marked as unsafe (#65918)

* Use is_sequence, and Mapping throughout, add support for tuples. Fixes #65722

* Address tests

* Remove unused import

* Add changelog

* Add docstring for clarity

* Argh, linting fix

* Not chasing this rabbit

* wrap_var doesn't return a ref to the original item

* no ref tests

* Remove unused import

Akasurde committed:
loop: simplify dict2items example in docs (#66235)

* Added a simple example to explain "dict2items" filters for better understading.

Fixes: #65505

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

sdodsley committed:
Fix small typo in purefa_host documentation examples (#66240)

mkrizek committed:
Move bitbucket unit tests to separate dir (#66237)

atombrella committed:
Stop using {{ }} in templates for conditionals in integration test (#65797)

Co-Authored-By: Matt Martz <matt@sivel.net>

nnhanndakara committed:
Update user module to support no_log=False. (#66219)

Updates the user module to support pull request #64733. Neither the
update_password or password_lock field contains sensitive information, so
mark them as such.

Qalthos committed:
Add test for reboot & wait_for_connection on EOS & IOS (#63014)

* Add test for reboot & wait_for_connection

* Add test for ios

* Collection-proof block test

* Add junos test

* Don't try to evaluate cli context unless using the connection

* Prevent infinite recursion

samccann committed:
Clarify how to use the Galaxy token (#65961)

* DOCS: clarify how to use the Galaxy token, with info for installs and api-key

mattclay committed:
Clean up Shippable scripts in tools directory.

pabelanger committed:
Add support for replacing an existing openstack keypair (#62762)

Signed-off-by: Paul Belanger <pabelanger@redhat.com>

Co-authored-by: Monty Taylor <mordred@redhat.com>

nkshrishail committed:
nxos: Removing local-nxapi plugin and related parameters' usage from the integration tests. (#65068)

* nxos_integration_tests: Removing obsolete provider parameter

* nxos: removing support for local-nxapi plugin

* nxos: removing LocalNxapi plugin implementation

* nxos_nxapi: removing local plugin usage

* REVERT - nxos: removing LocalNxapi plugin implementation

* Addressing yamllint issues

* Removing the condition on the debug messages

* Restoring changes done to nxos_nxapi* tests

* Retaining timeout of 500 on nxos_smoke test

* Some cleanup

* Addressing latest comments

* Restoring the nxos_smoke and removing timeout from nxos_config and nxos_rollback

dominikholler committed:
Fix typos in ios_l2_interfaces (#66246)

Some typos in the examples of ios_l2_interfaces are fixed.
This might help to have a fluent start with this module.

nkshrishail committed:
nxos_linkagg: removing unsupported parameter "provider" and "timeout" from sanity test (#64370)

* nxos_linkagg: removing timeout parameter from module nxos_feature

* nxos_linkagg: removing unsupported parameter "provider" from sanity test

* nxos_linkagg: typo correction

* nxos_linkagg: typo correction

goneri committed:
vmware_guest/test: create test vm in correct directory

Ensure we already create the `test_vm3` VM in the `{{ f0 }}` directory.
This to be sure the teardown playbook will be able to clean it up at the
end. This to avoid problem like this one:

```
Multiple virtual machines with same name [test_vm3] found, Folder value is a required parameter to find uniqueness of the virtual machine
```

Sara-Touqan committed:
Supporting bfd command in Ansible. (#64509)

Sara-Touqan committed:
Supporting AAA Configs in Ansible. (#64434)

Sara-Touqan committed:
Supporting SNMP Hosts Configurations in Ansible (#63983)

* Supporting SNMP Hosts Configurations in Ansible

* Modifying SNMP Hosts Configuration.

* Modifying SNMP Hosts Configurations.

* Modifying SNMP Hosts Configurations

* Modifying SNMP Hosts Configurations.

* Modifying SNMP Hosts Configurations Code.

Sara-Touqan committed:
New module: onyx_bfd (#64359)

* Supporting BFD in Ansible

* Modifying BFD module

alikins committed:
Don't remove trailing / on galaxy api URLS (#63294)

This causes multiple unneeded http redirects.
Since each http redirect takes a few hundred
milliseconds, handling the redirects starts to
take up a significant amount of wall time.

Fixes: #63281

mback2k committed:
Fix 'CyberarkPassword' object has no attribute 'delimiter' (#66268)

This is a very small follow up to PR #59500

mattclay committed:
Simplify docker_setup on Fedora.

This avoids installing dnf-plugins-core, which breaks the yum and dnf modules
when uninstalling packages using a wildcard after they have already been removed.

This should resolve issues with the yum integration tests failing after docker tests run.

rockaut committed:
adding proxy_address field for active proxies (#66247)

Since 4.0.0 there is a field proxy_address for limiting proxy requests.

Official Docs: "Proxy address	If specified then active proxy requests are only accepted from this list of comma-delimited IP addresses, optionally in CIDR notation, or DNS names of active Zabbix proxy.
This field is only available if an active proxy is selected in the Proxy mode field. Macros are not supported.
This option is supported since Zabbix 4.0.0."
updated documentation/help
fixed test issues
fixed version_added not a string
fix suggestions by D3DeFi


added changelog fragment
applied D3DeFi suggestions

way cleaner and better readable

heuels committed:
routeros_command – add support for dot in username (#65905)

jamescassell committed:
setup: more reliably detect container environment (#66183)

* virtual facts: /proc/<pid>/environ is NULL-separated
* non-empty "container" env is a guest container
* This is consistent with `systemd-detect-virt --container`
* fact utils: line_sep option: specify delim for split
* allow using the same function to split null-delimited objects

Andersson007 committed:
mysql_query: new module (#65941)

* tmp

* mysql_query: new module

* remove irrelevant changes from mysql.py

* add autocommit keyarg to mysql_connect

* add basic CI tests

* tmp

* fix CI tests

* add CI tests

* fix comment in CI tests

* remove path_to_script option

* refactoring

* change tests formatting

* add single_transaction param

* fix sanity

* add note

* fix

Akasurde committed:
FreeIPA: Fix documentation (#66194)

* Add datatype whereever it is missing
* Fix documentation

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

ganeshrn committed:
Fix net_put module text file issue (#66065)

*  Modify the parent class for net_put and net_get
   module to refer from `ActionModule` class in
   network.py action plugin which contains the
   common code for network plugins.
*  Replace `_handle_template` with `_handle_src_option`
   function name which is defined in common class.

rhysmeister committed:
mongodb_parameter module integration tests (#61644)

* Add final commits

* End of day check in

* Initial commit

* Remove file

* Remove module from pr

* Run tests

rhysmeister committed:
mongodb_user module integration tests (#61257)

* Add final commits

* End of night commit

* Add further tests

* What is this doing here?

* Remove file from commit

* Change os check

* Change version check

goneri committed:
vcenter provider: do not set env var

Some secrets are exposed because of the environment variables. The
Ansible variables are now enough, we don't need to expose the envvar
anymore.

goneri committed:
vmware_local_role_info/test: avoid json_query

Refactoring to avoid the use of `json_query`. As a bonus point, we now
also ensure `NoAccess` role has no provilege.

goneri committed:
vmware_vm_info/test: drop dep on json_query

Test refactoring to avoid dependency on `json_query`.

jborean93 committed:
ansible-galaxy ignore empty server_list (#65986)

* ansible-galaxy ignore empty server_list

yuandongx committed:
set up 'yanzhangi' as a reviewer for cloudengine. (#66024)

GomathiselviS committed:
Fix to nxos_vrf purge breaks with empty aggregate (#66004)

* resolving conflicts

* fixed issue with purge and state var. fixed space issue with vrf name

* lint issues

jborean93 committed:
Make sure warnings are printed during fact gathering (#66258)

tirkarthi committed:
Fix AttributeError due to cgi.escape removal in Python 3.8. (#66040)

* Fix cgi.escape AttributeError due to cgi.escape removal in Python 3.8.

* Fix linter errors.

jborean93 committed:
win_user_right - docs to show how to remove all (#66315)

nnhanndakara committed:
Allow no_log=False to silence the no_log warnings for module parameters (#64733)

As AnsibleModule._log_invocation is currently implemented, any parameter
with a name that matches PASSWORD_MATCH triggers the no_log warning as a
precaution against parameters that may contain sensitive data, but have not
been marked as sensitive by the module author.

This patch would allow module authors to explicitly mark the aforementioned
parameters as not sensitive thereby bypassing an erroneous warning message,
while still catching parameters which have not been marked at all by the
author.

Adds tests for various no_log states including True, False, and None (as
extracted by AnsibleModule._log_invocation) when applied to an argument with
a name that matches PASSWORD_MATCH.

Fixes: #49465 #64656

samdoran committed:
Do not install requests[security] for MongoDB tests

This was previously removed in PR ##60083 but was added again in
PR #61257. This will cause a lot of things to fail if it is uninstalled
during cleanup after the test role, which is why it was originally removed
from the test.

samdoran committed:
Add comment about not installing requests

samdoran committed:
Account for argument spec of None in validate modules sanity test

jborean93 committed:
Ansible.Basic - Added deprecated_aliases (#66323)

mattclay committed:
Add changelog entry for PR #66321

jborean93 committed:
Fix up deprecated_aliases spot for PS (#66326)

pabelanger committed:
Revert "Fix net_put module text file issue (#66065)" (#66328)

This PR didn't properly run testing for network jobs. As a result it has
broken jobs for ios and junos.

This reverts commit fa66813696b372bc4b8e0d836968e226a56928c7.

jborean93 committed:
WebRequest - remove generically named aliases from util (#66325)

* WebRequest - remove generically named aliases from util

* Fix location of deprecated aliases key

justjais committed:
updating metadata (#66289)

chrisvanheuveln committed:
nxos_l3_interfaces: fix states, add new minor attributes (#64853)

* (WIP) nxos_l3_interfaces: fix states, add new minor attributes

* sa cleanup

* more regression fixes

* test_8 for add'l code coverage

* Fix regressions to handle mgmt w/o IP

* add 'no system default switchport' to regression setups

* add err msg to terminal_stderr_re so that cli_config will catch L2 failures

* regression test change: /int4/int3/

* Add default rsvd_intf_len for Zuul CI

* Fix replaced-with-no-ipaddr and ip redirect issues

ganeshrn committed:
Add collection support for network platform agnostic modules (#66342)

*  Refactor net_base action plugin to invoke platform
   action plugin with implementation module name updated
   in the new task copy.

goneri committed:
vmware_vmotion/test: ensure we can run tests

Some adjustments to be able to run the test-suite properly:

- Starts with the ESXi out of the cluster to be able to deploy the
  VM on a proper host consistently
- Ensure the resource pool exists
- The resource pool is called `DC0_C0_RP1`, not `Resources`
- Avoid an exception if we try to move a non existing VM.

Put the test in the zuul/vmware/vcenter_2esxi group.

goneri committed:
vmware_category/test: ensure test can run on Zuul

- set the zuul/vmware/vcenter_only alias
- import `prepare_vmware_tests` role
- little refactoring to avoid an extra level of indentation

samccann committed:
added unit test details for resource modules (#65891)

* added unit test details for resource modules

samccann committed:
Document network resource modules (#66226)

* add network resource module details to docs

(cherry picked from commit 3e35d2d63ce17fb694e645785ad855f01420f7d8)

* network resource module page, with example output

samccann committed:
Docs: make different heading levels more obviously different in html output (#66253)

samccann committed:
add details on network gather facts improvements (#66220)

* add details on network gather facts improvements, implement feedback

samdoran committed:
Fix unarchive integration test on CentOS 6 (#66362)

* Cleanup packages added for Python TLS support

samccann committed:
Trim down to one Ansible search option  (#66284)

* move swiftype search to upper left of docsite, fix spacing and length issues for search icon and searchbox

billdodd committed:
Update deprecation version and porting guide for resource_id option (#66060)

* update deprecation version and porting guide for resource_id option

* add changelog fragment

goneri committed:
vmware_host_auto_start: enable the tests

- remove the unsupported alias
- refactoring
- move the hosts outside of the cluster to avoid any conflict with DRS
- import the `prepare_vmware_tests` role
- does not work with govcsim

mattclay committed:
Temporarily mark unarchive test unstable.

Qalthos committed:
Persistent connections: These timeouts seem unused (#66267)

* These timeouts seem unused

* Add changelog

psharkey committed:
new connection plugin aws_ssm (#49652)

* new connection plugin aws_ssm

Return code may be at the end of the last command output line.

Marking regex.

Ensure command status code is on it's owm line - last 3 lines are not part of command stdout.

* Adding timeout parameter - aws_ssm_timeout

Default 10 second timeout (https://docs.ansible.com/ansible/2.4/intro_configuration.html#timeout) is marginal.
This avoids changing this default and allowing the SSM timeout to be controlled via inventory.
This change wraps commands so commands which may never return do timeout.

* Added integration tests

Added AWS SSM Executor, target and config functions

Fixed more code for integration tests

Improved execution

Added S3 bucket name

Fixed pylint

Reverted lib changes

Reverted few more changes

Improved support for integration test execution

added ansible role for aws_ssm_integration_test setup and teardown and modifiled runme.sh

Reset to 17fa565 commit

inventory file location changed

change inventory file location

deleted meta and handlers folder as it is not required

deleted main.yml inside vars, removed extra space from tasks/main.yml, Added appropriate tags for ec2 and delete test folder as it is not required

deleted main.yml inside vars, removed extra space from tasks/main.yml, Added appropriate tags for ec2 and delete test folder as it is not required

modified task/main.yml

added region variable and fixed pattern for using variable

modified policy for IAM role

moved to first line of scrip set -eux

Updated Session Manager plugin installation

edited custme policy

Included tags for Session Manager plugin installation

Added README.md

Upddated README and added support for ssm-plugin for Amazon-Linux

Added Windows Integration test support

Improved user data for Linux

Added random value generation for the role and policy,delete vars_to_delete.yml

upadte README in vars

fixed typo

update policy

Updated IAM policy file

update playbook

Updated playbook to include ssm-agent userdata

modified jing2 template

modified jing2 template

modified jing2 template and fixed role deletion

fixed role name issue while deleting and task name

Updated playbook to include wait_connection for ec2

Corrected Synatx changes and updated ssm-plugin debian file

Changed region variable to us-east-1

Removed vars file and updated  to /tmp dir

fixed typo

Improved setup

Fixed boto3 dependency

Fixed missing tag

Added boto as dependency as well

Improved execution workflow

Trying other way of defining tags

Fixed undefined var

Changed AMI ID to Amazon Linux

Improved Tags

Ok, created different directory for WIndows test execution

Fixed IAM Role Name for Windows

Fixed inventory not found

Improved integration test execution

Fixed Windows Inventory path

Fixed wrong Windows AMI ID

Fixes issue for windows test execution

* Don't attempt to terminate sessions without a session id.

* Added Unit test cases file for AWS SSM Connection plugin

updated test file with close

updated unit test file with start_session

updated test files

* Eliminate AWS CLI dependency for terminal session.

* Removing unused code, cleanup logic.

Reduce mark length - 52^26 should be plenty
Be explicit about subprocess.Popen options
Simplify if/else for mark end
_stdin_readline is not used now

* updated test file

Added exec command and fixed close session unit tests

updated test files

Improved ansible ssm test command

updated file for lint checks

updated for pylint checks

New Unit_testcases for pre-signed URL file

removing additonal spaces and white spaces

remaning error changes

fixed changes

fixed spaces issues

python 2.7 version and whitespaces

python 2.7 version and whitespaces

python 2.7 skip if

space issue with 16:1

Unit test cases for windows and linux

Unit test cases for windows and linux with issues fixed issues

Unit test cases for windows and linux with issues fixed issues1

* Added support for S3 Pre-signed URLs

* Updated documentation and comments

* Documentation and curl dependency removal for controller machine

Fixing lint errors and removing requirements.

* Adding support for Windows remote EC2 instances.

* Added Encoding fixes

* Updating author section and adding obvious requirement for the SSM agent.

* Refactor stdout post processing
Attempt to get real return code on error (test using ansible -m raw -a 'cmd /c exit 99'.
Fixes problem at terminal width (ansible windows -i ./hosts.yml -m setup).

* Refactor back to a single module.

* Fixed fetch file for windows

* ssm usage examples for linux and windows

* Update aws_ssm.py

Service state corrected.

* Strip line continuation when at terminal width - otherwise replace.
Strip ANSI control sequences only for Windows.

Test playbook:

---
- name: test
  hosts: windows
  gather_facts: False
  vars:
    small: 'abc'
  tasks:
  - name:
    set_fact:
      large: "{{ lookup('password', '/dev/null length=2000 chars=ascii_letters,digits,hexdigits,punctuation') }}"
  - name: small fixed
    raw: echo '{{ small }}'
    register: small_result
    changed_when: False
  - name: check
    assert:
      that:
        - "(item | length) == (small | length)"
        - "item == small"
      msg: "'{{ item | length }} must equal '{{ small | length }}' and '{{ item }}' must match '{{ small }}'"
    with_items:
      - "{{ small_result.stdout_lines[0] }}"
  - name: large random
    raw: echo '{{ large }}'
    register: large_result
    changed_when: False
  - name: check
    assert:
      that:
        - "(item | length) == (large | length)"
        - "item == large"
      msg: "'{{ item | length }} must equal '{{ large | length }}' and '{{ item }}' must match '{{ large }}'"
    with_items:
      - "{{ large_result.stdout_lines[0] }}"
  - name: gather facts
    setup:

* Correct module parameter names.

* Updated Windows Executable variable.

Updated Windows Executable variable to "ansible_shell_type".

Fixing Examples with raw declaration

Updated the plugin timeout variable.

* Fix to work with dynamic inventory plug-in

* IntegrationTest template updates

* Removing unsupported flag for integration tests.
Fixing unit test.

* Adding shippable group.

* SSM Usage examples with dynamic inventory plugin

* Fixing yamllint errors.

* Fixed Integration tests

* Fixed Integration tests

* Updates for python3.

Removing python3 restriction.

* Remove python3 restriction.
Change block from retry to always

* Fixed Integration tests with Python 3

* Fixed shellcheck

* Fix for Windows which could pick up end mark prematurely
Move debug to _wrap_command and use a single return point

Single-quotes not needed around linux marks
Fix typo in comment

End mark to new command.

* Unit test cases now works on Python2 and Python3

* Skip tests on Python 2.6

* Fix for wait_for_connection module for windows.

* Updated changes as per review comments

* Fixing broken pipe error seen with session-manager-plugin version 1.1.17.0.
Eliminating sleep as this looks to be fixed in session-manager-plugin version 1.1.17.0.

* Adding back delays for Windows with session-manager-plugin 1.1.17.0.

* Updating Windows AMI ID for integration Test

* Upgrading windows ssm agent to the latest

* Adding boilerplate code.

* Windows ami and integration test updates

* Revert "Windows ami and integration test updates"

This reverts commit cd6ca3579b7cda584bd9c065f9c0835bddb23627.

* Updating windows ami for Integration tests

* Integration test suite updates and fixes.

* Updates and fixes

* Eliminate duplicate processing for exit code on failed command.

* Add powershell wrap.

* Refactor windows post_process.

* AMI Lookup, aliases, OSC filter, test suite updates

Co-authored-by: Gaurav Ashtikar <gau1991@gmail.com>
Co-authored-by: Deepak Choudhary <40276333+deepsvc@users.noreply.github.com>
Co-authored-by: Hanumanth <46720371+hanumantharaomvl@users.noreply.github.com>
Co-authored-by: KUMAR MAYANK <mayank@flux7.com>

Andersson007 committed:
Bugfix of 65367: postgresql_query doesn't support non-ASCII characters in SQL files with Python3 (#66331)

* Bugfix of 65367: postgresql_query doesn't support non-ASCII characters in SQL files with Python3

* add changelog

* fix

* change changelog fragment, add example

D3DeFi committed:
adjusted doc and argument spec in zabbix modules to match sanity requirements (#66176)

brlin-tw committed:
Clarify the advantage include* statement brought regarding looping (#66022)

Signed-off-by: 林博仁(Buo-ren, Lin) <Buo.Ren.Lin@gmail.com>

felixfontein committed:
openssl_* modules: allow direct input for some files (#66384)

* Allow to not read content from file.

* Allow to feed content directly into _info modules.

* Allow to feed non-primary content into openssl_certificate, openssl_csr and openssl_publickey.

* Rename changelog.

felixfontein committed:
Disable failing tests. (#66391)

felixfontein committed:
Fix aws_ssm connection plugin. (#66393)

Akasurde committed:
ec2_lc: Update example to use snapshot id (#66064)

Fixes: #52085

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

ganeshrn committed:
Add collection support with local for supported network platforms (#66302)

* Add collection support with local for supported network platforms

*  The legacy behavior of network action plugins using persistent
   the framework is to override the connection=local with the actual connection type (network_cli/netconf/httpapi) based
   on the value of transport option in provider.
*  After the actual connection plugin is identified try to load it from
   `ansible.netcommon` collection, if it fails load it from ansible
   core. This is done to work with Ansible 2.10 and also maintain backward compatibility for
   Ansible 2.9 version to ensure it works with network collections.
*  To support this need to pass collection_list value to connection
   plugins. As in case of connection=local it loads `persistent`
   connection within action plugin which in turn invokes `network_cli`
   or `netconf` connection. Similarly `network_cli` connection plugin invokes
   `terminal` and `cliconf plugins and `netconf` connection plugin invokes `netconf`
   `plugins` based on value on `ansible_network_os`.
*  Add deprecation warning for connection=local support and
   provider support for applicable platforms

* Fix CI failures

felixfontein committed:
Fix typo. (#66403)

mkrizek committed:
Check vars against Mapping in DEBUG (#66355)

When ANSIBLE_DEBUG=1, vars are VarsWithSources now which is a Mapping.
Check vars against Mapping instead of dict.

sky-joker committed:
Add proxy type passive example to example block  of zabbix_proxy (#66400)

ndclt committed:
Delete linter ignore lines for keycloak modules (#66406)

* linter: auth_client_id is not required, default value

Delete the lines linked to this mistake in ignore.txt.

* doc: state is not required

Qalthos committed:
net_facts modules: Don't warn about defaults changing if they're not being used (#66270)

goneri committed:
vcenter: don't bind container on <1024 port

Bind the govcsim service on port 1443, this way we don't need root privilege
to start the container. This allow us to use ansible-test with podman
seamlessly.
The commit also avoid the mapping of port 80. We don't need it.

xpvpc committed:
Update developing_python_3.rst (#66249)

##### SUMMARY

fix link § redirection

##### ISSUE TYPE
- Docs Pull Request

+label: docsite_pr

mattclay committed:
Add test constraint for setuptools. (#66426)

* Add test constraint for setuptools.

* Update pip test to work on centos6 container.

samccann committed:
added guidelines for seo; (#66431)

webknjaz committed:
Enforce new setuptools for building official dists

mattclay committed:
Ignore warnings in ansible-test environment check.

samdoran committed:
Fix unarchive integration test failure due to rpm error (#66420)

* Fix unarchive integration test failure
* Do not cleanup pip packages

stawii committed:
ec2_vol module filtering bug (#65960)

* Update ec2_vol.py

When `name` is specified, it ignores `zone` filter. That is you could not have same tag:Name in different zones.

* Add changelog fragment

jborean93 committed:
Deprecate Windows 2008, and 2008 R2 (#66257)

* Deprecate Windows 2008, and 2008 R2

* Remove shippable nodes

* Update windows_faq.rst

Be less specific about 2008/R2 timeframes

* Update setup.ps1

tweak warning text

Co-authored-by: Matt Davis <nitzmahone@users.noreply.github.com>

jpmens committed:
make: use -f instead of --file to aid in portability (#65880)

`make` on FreeBSD (and others) has support for an option `-f` to specify a `Makefile`, whereas GNU make uses `--file` or `-f`. In the interest of portability use `-f` which covers both.

s-hertel committed:
RoleRequirement - Include stderr in the error message (#66006)

* RoleRequirement - Include stderr in the error message if there's a non-0 return code

* Don't try to concatenate str and bytes

Co-Authored-By: Sam Doran <sdoran@redhat.com>

ad1rie1 committed:
Fix Bug (#64553)

adharshsrivatsr committed:
Bug fix for #53271 (return commands for cli_config) (#61808)

* cli_config return commands bug fix

* Spacing error

* Added notes section

* Spacing error

* Spacing error

* Spacing error

* Spacing error

goneri committed:
prepare_vmware_tests: no waiting loop during DS probing

For a refresh of the datastore list on the vcenter, this before we
check the present of the newly attached datastores.

A lot of tests depend on the presence of the DS, and this little
change allow us to save =~ 45s everytime.

Qalthos committed:
Allow httpapi for resource modules (#62843)

* Redo tests to be transport agnostic

cli -> eos config

Redirect connection for httpapi

ric-man committed:
terraform: Remove line that is suppressing output being shown (#66322)

* remove the line that is suppressing the output being shown when running
terraform from ansible
* Address #56934 and #57044
* added changelog for bug fix for missing terraform output

Co-authored-by: Adam <adam.lemanski@gmail.com>

Commit:
vmware_inventory: define si outside try/except (#66482)

goneri committed:
vmware_guest_serial_port: avoid autoselect_datastore (#66481)

Avoid `autoselect_datastore` until #58541 gets fixed.

Akasurde committed:
Fortios: Correct underscore_to_hyphen API (#66119)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

xiaopeng163 committed:
add one note to help ansible user to understand the requirements to (#65969)

execute iosxr_command module

Signed-off-by: Peng Xiao <xiaoquwl@gmail.com>

Akasurde committed:
VMware: Use existing DVPG network in vmware_guest_network (#65994)

* Handle all cases of networks

Fixes: #65968

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

pabelanger committed:
Revert "Allow httpapi for resource modules (#62843)" (#66483)

While this does properly pass our testing for ansible/ansible devel
branch, it is currently breaking our collection testing for 2.10.

Specifically, this would mean ansible.netcommon would need to directly
import arista.eos or cisco.nxos collections, causing a circular dependency.

This reverts commit e266e5f8b64c344fe7b9d711afe11c32e07954fa.

s-hertel committed:
Add a metadata property for CollectionRequirement (#66370)

* Add a metadata property for CollectionRequirement so the _metadata attribute doesn't need to be accessed directly

* changelog

Commit:
podman_image: Use provided credentials when pulling image (#66372)

* Use credentials when pulling from a remote registry

evgeni committed:
use `query` instead of `q` in "if files exist" example (#66409)

##### SUMMARY
It's not obvious that `q` is the same as `query` (which is referred to in the name of the task) and using long-names is generally preferred in examples.

##### ISSUE TYPE
- Docs Pull Request

+label: docsite_pr

C0rn3j committed:
Update and cleanup Arch packaging (#64747)

hiyokotaisa committed:
Fixes #65590 add pywinrm version explanation (#65591)

* add pywinrm version explanation

* Update windows_winrm.rst

Changed pywinrm explanation to note style

* Add period

Co-authored-by: Jordan Borean <jborean93@gmail.com>

jborean93 committed:
Use UTF-8 without a BOM for win coverage (#66510)

haiyuazhang committed:
Vmss scale in policy (#66512)

* azure_rm_virtualmachinescaleset: add scale_in_policy and terminate_event_notification

* azure_rm_virtualmachinescalesetinstance: add vmss instance protection policy support

lassik committed:
hostname - Add Void Linux distribution (#66499)

Add support for Void Linux in hostname module.

jainnikhil30 committed:
Foreman: support for want_collectionhosts and want_ansible_ssh_host

* Add support for want_collectionhosts and want_ansible_ssh_host in foreman plugin

acalm committed:
Add diff and check_mode support to os_server (#62206)

* don't mix " and '

* rename functions that perform changes

* add diff, reorganize _present/_absent functions

* update tests to match function reorg

* add result to _exit_hostvars and removal of possibly sensitive information

* shallow copy dict for proper diff, add check_mode

* set check_mode to False for unit tests

* move _parse_meta back to it's original place

* use get_openstack_vars for diff when not in check_mode

* add changelog fragment

pabelanger committed:
Fix flake8 errors related to vyos.vyos collection (#66516)

Signed-off-by: Paul Belanger <pabelanger@redhat.com>

samccann committed:
restructure headings and copy-edit shell completion (#66479)

* restructure headings and fix shell completion, apply suggestions from code review

Co-Authored-By: Alicia Cozine <879121+acozine@users.noreply.github.com>

GomathiselviS committed:
eos_lacp_interfaces: Add unit testcases to eos_lacp_interfaces (#64455)

* Add unit testcases to eos_lacp_interfaces

* Add unit testcases to eos_lacp_interfaces

samdoran committed:
ansible-galaxy - add "--token" command line argument (#66376)

bandit145 committed:
fixed documentation issue

bandit145 committed:
Merge branch 'semodule' of github.com:bandit145/ansible into semodule

bandit145 commented:
There is an issue with the current opensuse15 integration test container, the container is 15.0 and it looks like the 15.0 SELinux repository is gone. I did try the 15.1 repo on opensuse 15.0 which did work but that would mean hard coding the 15.1 repository into the test and that would be really brittle in the future.

I'm leaning towards disabling the opensuse15 integration test but if the container is going to be updated soon I won't.

15.1 repo: https://download.opensuse.org/repositories/security:/SELinux/openSUSE_Leap_15.1/security:SELinux.repo

15.0 repo: https://download.opensuse.org/repositories/security:/SELinux/openSUSE_Leap_15.0/security:SELinux.repo
